libgstc/pygstc/tcp.py

```diff
@@ -127,45 +148,41 @@ def _recvall(self, sock, timeout):
         timeout : int
             Timeout in seconds to wait for a response. 0: infinite
 
+        Raises
+        ------
+        socket.error
+            Error is triggered when Gstd IPC fails
+        BufferError
+            When the incoming buffer is too big.
+
         Returns
         -------
         buf : string
             Raw socket response
         """
-
         buf = b''
-        count = 0
+        newbuf = ''
         try:
-            while True:
-                if self._maxsize:
-                    if count >= self._maxsize:
-                        break
-
-                # if timeout is set, perform non-blocking read
-
-                if timeout:
-                    sock.setblocking(0)
-                    ready = select.select([sock], [], [], timeout)
-                    if ready[0]:
-                        newbuf = sock.recv(self._socket_read_size)
-                    else:
-                        buf = None
-                        sock.close()
-                        break
-                else:
+            sock.settimeout(timeout)
+        except socket.error as e:
+            raise TimeoutError from e
+
+        while True:
+            if (self._maxsize and self._maxsize > len(newbuf)):
+                raise BufferError
+            # Timeout to perform non-blocking read
+            ready = select.select([sock], [], [], timeout)
```

Comment:
I'm not convinced this is still necessary.

Reply:
Otherwise:
`socket.error`
On:
`newbuf = sock.recv(self._socket_read_size)`